tests/phpunit/integration/MediaWiki/Api/BatchEditEndpointTest.php

```diff
@@ -0,0 +1,263 @@
+<?php
+
+namespace MediaWiki\Extension\WikibaseReconcileEdit\Tests\Integration\MediaWiki\Api;
+
+use MediaWiki\Extension\WikibaseReconcileEdit\InputToEntity\MinimalItemInput;
+use MediaWiki\Extension\WikibaseReconcileEdit\MediaWiki\Api\BatchEditEndpoint;
+use MediaWiki\Extension\WikibaseReconcileEdit\MediaWiki\Request\EditRequestParser;
+use MediaWiki\Extension\WikibaseReconcileEdit\MediaWiki\WikibaseReconcileEditServices;
+use MediaWiki\Extension\WikibaseReconcileEdit\Reconciliation\ItemReconciler;
+use MediaWiki\Rest\LocalizedHttpException;
+use MediaWiki\Rest\RequestData;
+use MediaWiki\Rest\RequestInterface;
+use MediaWiki\Session\SessionProviderInterface;
+use MediaWiki\Tests\Rest\Handler\HandlerTestTrait;
+use User;
+use Wikibase\DataModel\Entity\Item;
+use Wikibase\DataModel\Entity\ItemId;
+use Wikibase\DataModel\Entity\Property;
+use Wikibase\DataModel\Entity\PropertyId;
+use Wikibase\DataModel\Snak\PropertyValueSnak;
+use Wikibase\Repo\WikibaseRepo;
+
+/**
+ * @covers MediaWiki\Extension\WikibaseReconcileEdit\MediaWiki\Api\BatchEditEndpoint
+ * @group Database
+ * @license GPL-2.0-or-later
+ */
+class BatchEditEndpointTest extends \MediaWikiIntegrationTestCase {
```

Comment:
This reminds me a lot of EditEndpoint test, not sure there are any real benefits of doing so but maybe we could refactor these afterwards to share some of this code / token tests etc if we feel there is enough time.